### PR TITLE
Create 'Provide a good engineering experience' principle

### PR DESCRIPTION
**Pull request by edhamiltonHO**
# Content change 
I can confirm:
- [x] Content does not include any code or configuration changes (excluding frontmatter information)
- [x] Content meets the content standards
e.g. [Writing a principle](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-principle/) and [Writing a standard](https://ho-cto.github.io/engineering-guidance-and-standards/docs/standards/writing-a-standard/)
- [x] Content is suitable to open source, i.e.:
    - Content does not relate to unreleased gov policy
    - Content does not refer to anti-fraud mechanisms
    - Content does not include sensitive business logic

### CODE REVIEW

**Comment by ezramorris** [comment]
I think a title like "Provide a good engineering experience" would be better, as "improve" is difficult to measure and doesn't suggest a baseline.

File: docs/principles/provide-good-engineering-experience.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+It is important to remember that engineers are users too, we already have proven patterns for the best ways to meet user needs and they should be followed. To improve engineer experience we will meet engineering needs in the same way as we would with any other user need, centering engineers' feedback and experience when testing solutions that have been designed for them. This will provide the Home Office with a measurable and evidenced approach to factors that contribute to productivity and efficiency.
+
+---
+
+## Applications and Implications
+
+- Approach cloud platform engineering in a way that delivers platforms as products that are developed according to tenants needs
+- Do user research with engineers to understand how they use tools and services, and where we can improve
```
Comment (GraemeSutherlandHO) [inline docs/principles/provide-good-engineering-experience.md:30]: remove 'user' here. avoid conflation/confusion with big U User Research which I think is different

File: docs/principles/provide-good-engineering-experience.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+It is important to remember that engineers are users too, we already have proven patterns for the best ways to meet user needs and they should be followed. To improve engineer experience we will meet engineering needs in the same way as we would with any other user need, centering engineers' feedback and experience when testing solutions that have been designed for them. This will provide the Home Office with a measurable and evidenced approach to factors that contribute to productivity and efficiency.
+
+---
+
+## Applications and Implications
+
+- Approach cloud platform engineering in a way that delivers platforms as products that are developed according to tenants needs
```
Comment (GraemeSutherlandHO) [inline docs/principles/provide-good-engineering-experience.md:29]: not sure about the word Tenant here.. Either this is about engineers, or is about tenants, so maybe this point is in platform principles not here?

File: docs/principles/provide-good-engineering-experience.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Last updated: {{ page.date | postDate }}
+{{ pageTag.tags(tags)  }}
+
+At the Home Office, we recognise that the [engineer experience](https://github.blog/2023-06-08-developer-experience-what-is-it-and-why-should-you-care/) is comprised of many different aspects, and that getting those things right allows engineers to produce their best work.  We will invest in the continuous improvement of our engineering experience.
+
+---
+
+## Rationale
+
+The Home Office engineering experience is more than tooling and platforms and technology meeting engineers' needs. It is affected by our working relationships, our community, our structures and our culture. Engineers complete work more quickly, and at higher quality, when the tools, environments and teams they work with provide the right capabilities and support. When they don't have those things, work moves more slowly, technical debt is accrued and security can be compromised. We want to reduce day to day friction and duplication, to improve productivity.
```
Comment (jeff-horton-ho-sas) [inline docs/principles/provide-good-engineering-experience.md:21]:
```suggestion
The Home Office engineering experience is more than tooling and platforms and technology meeting engineers' needs. It is affected by our working relationships, our community, our structures and our culture. Engineers complete work quicker and at higher quality when the tools, environments and teams they work with provide the right capabilities and support. When they don't have those things, work moves more slowly, technical debt is accrued and security can be compromised. We want to reduce day-to-day friction and duplication, to improve productivity.
```

File: docs/principles/provide-good-engineering-experience.md (hunk 1 above)
Comment (jeff-horton-ho-sas) [reply docs/principles/provide-good-engineering-experience.md:30]: I would argue that the engineers are the users of the tools and services and using user research methodology is appropriate.

File: docs/principles/provide-good-engineering-experience.md (hunk 2 above)
Comment (jeff-horton-ho-sas) [reply docs/principles/provide-good-engineering-experience.md:29]: Agree this should be written from an engineer's perspective, maybe:
```suggestion
- The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience.
```

File: docs/principles/provide-good-engineering-experience.md (hunk 1 above)
Comment (edhamiltonHO) [reply docs/principles/provide-good-engineering-experience.md:30]: Accept that the sentence needs to be clarified, but agree with Jeff on the intent. Will reword

File: docs/principles/provide-good-engineering-experience.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+It is important to remember that engineers are users too, we already have proven patterns for the best ways to meet user needs and they should be followed. To improve engineer experience we will meet engineering needs in the same way as we would with any other user need, centering engineers' feedback and experience when testing solutions that have been designed for them. This will provide the Home Office with a measurable and evidenced approach to factors that contribute to productivity and efficiency.
+
+---
+
+## Applications and Implications
+
+- The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience.
```
Comment (robertdeniszczyc2) [inline docs/principles/provide-good-engineering-experience.md:29]:
```suggestion
- The Home Office's cloud platform engineering should deliver platforms as products that are developed according to tenant projects' needs. Engineers should engage with platforms to provide the feedback needed to continuously improve the developer experience
```

So it is consistent with the other bullet points, remove the full stop

**Review by RobJohnson-HO — APPROVED** [review]
LGTM

**Pull request merged**